integration/e2e/metrics_promql.go

```diff
@@ -0,0 +1,283 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/storage"
+)
+
+var (
+	errTooManyVectorElements = "query returned vector with %d values"
+	errNotScalarOrVector     = errors.New("not a scalar or vector with single value")
+)
+
+func evaluatePromQLWithTextMetrics(expr string, metrics string) (float64, error) {
+	var tp expfmt.TextParser
+	families, err := tp.TextToMetricFamilies(strings.NewReader(metrics))
+	if err != nil {
+		return 0, err
+	}
+
+	now := time.Now()
+
+	samples := convertMetricFamiliesToSamples(int64(model.TimeFromUnixNano(now.UnixNano())), families)
+
+	eng := promql.NewEngine(promql.EngineOpts{
+		Logger:             nil,
+		Reg:                nil,
+		MaxSamples:         1e6,
+		Timeout:            time.Hour, // time.Second,
```

Comment:
There's some commented code here. I guess few seconds timeout is enough.

Reply:
Leftover from manually stepping through the code and hitting timeouts 😄